### ISSUE TITLE
Finer tuning of split separator?

### ISSUE THREAD

**Issue by NamesMT** [blakeembrey/change-case#319]
Currently change-case parses (splits) multiple types of casing in one pass like:

Using `capitalCase()`:
`Hello from Sn_ake's iPhone` => `Hello From Sn Ake S I Phone`
`Check out: 101 iAndroid` => `Check Out 101 I Android` ## We are also missing the colon here

It would be amazing if we could have some / an option to finer tuning to achive results like:
`Hello From Sn_ake's IPhone` ## white space only
`Hello From Sn_ake's iPhone` ## detect common noun styling ([a-z][A-Z])

We could also think about export the factory functions so that user could use custom split functions, or accept a split function as an option.

**Comment by blakeembrey** [comment]
Can you describe more about what you’re actually trying to do? I don’t think this library is designed to fit your needs, you’re probably better off just writing the code to do whatever it is you want yourself.

**Comment by NamesMT** [comment]
Hi Blake,
TL;DR of what I'm trying to achieve is a way to make the split function less agressive, or being able to pass a custom split function to it 😄
The library is awesome but sometimes too agressive in some cases as mention in the first post, that's all.

**Comment by blakeembrey** [comment]
The issue with your example is it still won’t work as expected, the case changing functions also lower case other characters. 

It seems like what you really want is just the existing title case library? It doesn’t do any splitting or white space changes and avoids messing with words like iPhone.

**Comment by NamesMT** [comment]
Kk thank you for your time,
My network have been a pain to browse around and I didn't actually visit the factory functions,
It seems like a new package is easier to fit my case than commitings
